### PR TITLE
feature/PEER-764 : Correct data in documentation articles

### PR DESCRIPTION
**Pull request by Dmitrijg24k**

### CODE REVIEW

File: app/components/Documentation/EditDocumentation.tsx [hunk 1]
```diff
@@ -226,6 +230,8 @@ const EditDocumentation: React.FC<EditDocumentationProps> = ({
                   <ViewContent
                     documentationArticle={viewDocumentationArticle}
                     isEditDocumentation
+                    locale={locale}
+                    isEditPost={!!isDraft}
```
Comment (Lampardks) [inline app/components/Documentation/EditDocumentation.tsx:234]: isEditPost={Boolean(isDraft)}

File: app/components/Documentation/components/Drafts/Drafts.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ type DraftsMenuSectionProps = {
   draftsMenu: Array<DocumentationSection>;
   setEditArticle?: (data: EditArticleType) => void;
   setViewArticle?: (id: string) => void;
-  draftsIds: Array<string>;
+  draftsIds: { draftId: string; lastmod: string }[];
```
Comment (Lampardks) [inline app/components/Documentation/components/Drafts/Drafts.tsx:13]: draftsIds: Array<{ draftId: string; lastmod: string }>;

File: app/components/Documentation/components/Drafts/DraftsItem.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ type DraftsItemProps = {
   parentId: string;
   setEditArticle?: (data: EditArticleType) => void;
   setViewArticle?: (id: string) => void;
-  draftsIds: Array<string>;
+  draftsIds: { draftId: string; lastmod: string }[];
```
Comment (Lampardks) [inline app/components/Documentation/components/Drafts/DraftsItem.tsx:13]: draftsIds: Array<{ draftId: string; lastmod: string }>;

File: app/components/Documentation/components/ViewContent.tsx [hunk 4]
```diff
@@ -13,21 +19,34 @@ const styled = {
...
   documentationArticle,
   isEditDocumentation,
+  locale,
+  isEditPost,
 }): JSX.Element => {
   const headers = extractStrings(['#', '\n'])(
     `${documentationArticle?.content}\n` || '',
   );
-
+  console.log(123, isEditDocumentation);
```
Comment (Lampardks) [inline app/components/Documentation/components/ViewContent.tsx:49]: delete

File: app/components/Documentation/helpers.tsx [hunk 5]
```diff
@@ -26,20 +26,30 @@ export const initMenu = (
   },
 ];
 
-export const getSavedDraftsIds = (): Array<string> => {
+export const getSavedDraftsIds = (): { draftId: string; lastmod: string }[] => {
```
Comment (Lampardks) [inline app/components/Documentation/helpers.tsx:29]: Array<{ draftId: string; lastmod: string }>

File: app/components/Documentation/helpers.tsx [hunk 6]
```diff
@@ -26,20 +26,30 @@ export const initMenu = (
...
   if (localStorage.getItem(DRAFTS_IDS)) {
-    return JSON.parse(localStorage.getItem(DRAFTS_IDS));
+    const draftIds = localStorage.getItem(DRAFTS_IDS);
+    if (draftIds) {
+      const draftIdsArray = JSON.parse(draftIds);
+      return draftIdsArray;
```
Comment (Lampardks) [inline app/components/Documentation/helpers.tsx:34]: return JSON.parse(draftIds);

File: app/components/Documentation/helpers.tsx [hunk 7]
```diff
@@ -26,20 +26,30 @@ export const initMenu = (
...
   }
 
   return [];
 };
 
-export const saveDraftsIds = (draftId: string): Array<string> => {
+export const saveDraftsIds = (
+  draftId: string,
+  lastmod: string,
+): { draftId: string; lastmod: string }[] => {
```
Comment (Lampardks) [inline app/components/Documentation/helpers.tsx:44]: Array<{ draftId: string; lastmod: string }>

File: app/pages/Documentation/saga.ts [hunk 8]
```diff
@@ -33,14 +38,33 @@ export function* getArticleDocumentationWorker({
...
-
-      yield put(
-        getArticleDocumentationSuccess({
-          id: articleId,
-          ...JSON.parse(documentationArticle as string),
-        }),
-      );
+      const draftsIds = yield select(selectDraftsIds());
+      const editedPost = (
+        draftsIds as { draftId: string; lastmod: string }[]
```
Comment (Lampardks) [inline app/pages/Documentation/saga.ts:43]: Array<{ draftId: string; lastmod: string }>

File: app/components/Documentation/EditDocumentation.tsx (hunk 1 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/EditDocumentation.tsx]: +

File: app/components/Documentation/components/Drafts/Drafts.tsx (hunk 2 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/components/Drafts/Drafts.tsx]: +

File: app/components/Documentation/components/Drafts/DraftsItem.tsx (hunk 3 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/components/Drafts/DraftsItem.tsx]: +

File: app/components/Documentation/components/ViewContent.tsx (hunk 4 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/components/ViewContent.tsx]: +

File: app/components/Documentation/helpers.tsx (hunk 5 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/helpers.tsx]: +

File: app/components/Documentation/helpers.tsx (hunk 6 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/helpers.tsx]: +

File: app/components/Documentation/helpers.tsx (hunk 7 above)
Comment (Dmitrijg24k) [reply app/components/Documentation/helpers.tsx]: +

File: app/pages/Documentation/saga.ts (hunk 8 above)
Comment (Dmitrijg24k) [reply app/pages/Documentation/saga.ts]: +

**Review by Lampardks — APPROVED** [review]

**Pull request closed without merging**